projects/2023-NH/analysis.script_hadacek.R

```diff
@@ -38,74 +38,114 @@ sessionInfo()
 
-#creating self conscientiousness variable
+# creating self conscientiousness variable 
 consci_items_self <- dict_self %>% 
   filter(scale == "BFAS Conscientiousness") %>% 
   pull(variable)
 data_self$consci_self <- data_self %>% 
   select(all_of(consci_items_self)) %>% 
   rowMeans(na.rm = TRUE)
 
-#creating pid rigid perfectionism variable
+# creating pid rigid perfectionism variable 
 pid_items <- dict_self %>% 
   filter(scale == "PID Miscellaneous") %>% 
   pull(variable)
 data_self$pid <- data_self %>% 
   select(all_of(pid_items)) %>% 
   rowMeans(na.rm = TRUE)
 
-#creating pil variable
+# creating pil variable 
 pil_items <- dict_self %>% 
   filter(scale == "PIL Purpose in Life") %>% 
   pull(variable)
 data_self$pil <- data_self %>% 
   select(all_of(pil_items)) %>% 
   rowMeans(na.rm = TRUE)
 
-#creating neuroticism variable
+# creating neuroticism variable 
 neuro_items <- dict_self %>% 
   filter(scale == "BFAS Neuroticism") %>% 
   pull(variable)
 data_self$neuro <- data_self %>% 
   select(all_of(neuro_items)) %>% 
   rowMeans(na.rm = TRUE)
 
-#creating withdrawal variable
+# creating withdrawal variable 
 withdraw_items <- dict_self %>% 
-  filter(scale == "BFAS Withdrawal") %>% 
+  filter(subscale == "BFAS Withdrawal") %>% 
   pull(variable)
 data_self$withdraw <- data_self %>% 
   select(all_of(withdraw_items)) %>% 
   rowMeans(na.rm = TRUE)
 
-#merging dataframes
+# merging dataframes 
 jointdataset <- merge(data_self, peer_group, by = 'ID')
 
-#creating conscientiousness difference score
+# creating conscientiousness difference score 
 jointdataset <- jointdataset %>% 
   dplyr::mutate(consci_dif = (consci_peer - consci_self))
+
+# REGRESSION ANALYSIS ####
+
+## simple regression ####
+simple_regression <- lm(consci_dif ~ pid, data = jointdataset)
+summary(simple_regression)
+#interpreting
+  #consci_dif=(-.25410)pid+.68551
+  #p=1.523e-06
+  #For every increase in rigid perfectionism of 1, the difference between 
+  #peer-reported conscientiousness and self-reported conscientiousness decreases 
+  #by -0.2541. Given that rigid perfectionism ranges from 1-4 and 
+  #conscientiousness ranges from 1-5, this is a moderate-sized correlation 
+  #(r=-0.3189428). The result is also statistically significant (p<.05).
+
+#visualize
+ggplot(jointdataset, aes(pid, consci_dif)) +
+  geom_point(color = "#f1c232") +
+  geom_smooth(method = "lm", color = "#710c0c") +
+  labs(title = "Multiple Regression", x = "Rigid Perfectionism", 
+       y = "Conscientiousness Difference Score")
+
+#correlation test
+cor(jointdataset$consci_dif, jointdataset$pid, use = "pairwise")
+
+## multiple regression ####
+#multiple_regression <- lm(outcome_variable ~ pred1 + pred2 + pred.etc, data = data)
+#for moderator, use * instead of +
+
+multiple_regression <- lm(consci_dif ~ pid + pil + neuro + withdraw, 
+                          data = jointdataset)
+summary(multiple_regression)
+#interpreting
+  #consci_dif=(-.33045)pid+1.79231
```

Comment:
the formula for multiple regression actually includes all the other coefficients.

for example: let a be the coefficient (slope) for pid, b be the coefficient for pil, c for the coefficient for neuro, and d for withdraw -> the formula is:
consci_dif = a*pid + b*pil + c*neuro + d*withdraw + intercept